packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -247,42 +248,62 @@ export async function scanParentDirs(
     readPackageJson && pkgJsonPath
       ? JSON.parse(await fs.readFile(pkgJsonPath, 'utf8'))
       : undefined;
-  const [yarnLockPath, npmLockPath, pnpmLockPath] = await walkParentDirsMulti({
-    base: '/',
-    start: destPath,
-    filenames: ['yarn.lock', 'package-lock.json', 'pnpm-lock.yaml'],
-  });
+
   let lockfilePath: string | undefined;
   let lockfileVersion: number | undefined;
-  let cliType: CliType = 'yarn';
-
-  const [hasYarnLock, packageLockJson, pnpmLockYaml] = await Promise.all([
-    Boolean(yarnLockPath),
-    npmLockPath
-      ? readConfigFile<{ lockfileVersion: number }>(npmLockPath)
-      : null,
-    pnpmLockPath
-      ? readConfigFile<{ lockfileVersion: number }>(pnpmLockPath)
-      : null,
-  ]);
-
-  // Priority order is Yarn > pnpm > npm
-  if (hasYarnLock) {
-    cliType = 'yarn';
-    lockfilePath = yarnLockPath;
-  } else if (pnpmLockYaml) {
-    cliType = 'pnpm';
-    lockfilePath = pnpmLockPath;
-    lockfileVersion = Number(pnpmLockYaml.lockfileVersion);
-  } else if (packageLockJson) {
-    cliType = 'npm';
-    lockfilePath = npmLockPath;
-    lockfileVersion = packageLockJson.lockfileVersion;
+  let cliType: CliType | undefined;
+
+  if (corepackEnabled && readPackageJson) {
```

Comment:
Not sure about this `readPackageJson` boolean.

I'm starting to think it always needs to read or else corepack won't be detected, right?

Reply:
this function is doing too much IMO. 
it seems to be called in some places purely to read the package.json, and in other places to do the cliType and lockfile resolution. maybe splitting it up would be better

Reply:
Yeah I think the cases that only care about package.json could call `walkParentDirs()` directly